### PR TITLE
Add CI for the PRs

### PR DESCRIPTION
**Pull request by krzysztofwolski**
- migrate linters and CI from the sdk to force same code style
- add CI jobs and hooks
- a bunch of smaller code changes to satisfy eslint

### CODE REVIEW

File: components/AccessWarning/AccessWarning.tsx [hunk 1]
```diff
@@ -18,15 +18,17 @@ const warnings: Record<WarningCause, string> = {
...
-    <Typography variant="subtitle1">
-      App can&apos;t be accessed outside of the Saleor Dashboard
-    </Typography>
-    <Typography variant="subtitle2" style={{ marginTop: "2rem" }}>
-      ❌ {warnings[cause]}
-    </Typography>
-  </div>
-);
+function AccessWarning({ cause = "unknown_cause" }: AccessWarningProps) {
+  return (
```
Comment (Cloud11PL) [inline components/AccessWarning/AccessWarning.tsx:22]: Why are we using the old `function xyz...` format?

Comment (Cloud11PL) [reply components/AccessWarning/AccessWarning.tsx:22]: Please let's not 🙈

Comment (krzysztofwolski) [reply components/AccessWarning/AccessWarning.tsx:22]: @Cloud11PL it's a style guide from the https://github.com/saleor/react-storefront/

Comment (Cloud11PL) [reply components/AccessWarning/AccessWarning.tsx:22]: Are we sure we want it in our template though?

Comment (krzysztofwolski) [reply components/AccessWarning/AccessWarning.tsx:22]: Don't have a strong opinion regarding this rule, but would like to have consistency between our projects.

**Pull request merged**